packages/gatsby/src/utils/create-node-id.js

```diff
@@ -11,6 +12,12 @@ const seedConstant = `638f7a53-c567-4eca-8fc1-b23efb1cfb2b`
  * @return {String} - UUID
  */
 function createNodeId(id, namespace) {
+  if (typeof id === `number`) {
+    id = id.toString()
+  } else if (typeof id !== `string`) {
+    report.panic(`Parameter passed to createNodeId must be a String or Number`)
```

Comment:
maybe we could display actual type that was passed to give more context?
```suggestion
    report.panic(`Parameter passed to createNodeId must be a String or Number (got ${typeof id})`)
```